src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -384,6 +387,28 @@ private List<E> queryNextBlock() {
 
             return effectiveQuery.where(OMA.FILTERS.gt(leftHandSide, rightHandSide)).queryList();
         }
+
+        private static Object getPropertyValue(EntityDescriptor entityDescriptor,
+                                               Mapping mapping,
+                                               Object entity) {
+            if (mapping.getParent() != null) {
+                Object parent = getPropertyValue(entityDescriptor, mapping.getParent(), entity);
+                return getPropertyValue(mixing.getDescriptor(parent.getClass()),
+                                        Mapping.named(mapping.getName()),
+                                        parent);
+            }
+            Property property = entityDescriptor.getProperty(mapping.getName());
+            if (property instanceof BaseEntityRefProperty<?, ?, ?> ref) {
+                BaseEntityRef<?, ?> entityRef = ref.getEntityRef(entity);
+                return entityRef.getValueIfPresent().orElseThrow(() -> {
+                    return new IllegalArgumentException(Strings.apply(
+                            "The BaseEntityRef `%s` is not loaded, but is requested by the mapping `%s`.",
+                            entityRef.getUniqueObjectName(),
+                            mapping));
+                });
+            }
```

Comment:
isn't this only effective within the parent lookup(s) - can't we move this into an extra method? this whole method seems to do two things at once..